src/V2/player.ts

```diff
@@ -0,0 +1,461 @@
+            for (; currentRequestingFrame <= this.lastRequestedGeometryFrame; currentRequestingFrame++) {
+                const padWidth = this.countHashChar(this.fileHeader.DRCURLPattern);
+                const dracoURL = this.fileHeader.DRCURLPattern.replace('#'.repeat(padWidth), this.pad(currentRequestingFrame, padWidth));
+                promises.push(this.decodeDraco(dracoURL, currentRequestingFrame));
+            }
+        }
+        if ((this.lastRequestedTextureSegment - this.currentTextureSegment) < this.textureBufferSize && this.lastRequestedTextureSegment != (this.fileHeader.TextureSegmentCount - 1)) {
+            let currentRequestingTextureSegment = this.lastRequestedTextureSegment + 1;
+            this.lastRequestedTextureSegment = Math.min(this.currentTextureSegment + this.textureBufferSize, this.fileHeader.TextureSegmentCount - 1);
+            for (; currentRequestingTextureSegment <= this.lastRequestedTextureSegment; currentRequestingTextureSegment++) {
+                const padWidth = this.countHashChar(this.fileHeader.KTX2URLPattern);
+                const textureURL = this.fileHeader.KTX2URLPattern.replace('#'.repeat(padWidth), this.pad(currentRequestingTextureSegment, padWidth));
+                promises.push(this.decodeKTX2(textureURL, currentRequestingTextureSegment));
+            }
+        }
+
+        if (callback) {
+            Promise.all(promises).then(() => {
+                callback()
+            })
+        }
+    }
+
+    decodeDraco = (dracoURL: string, frameNo: number) => {
+        return new Promise((resolve, reject) => {
+            this.dracoLoader.load(dracoURL, (geometry: BufferGeometry) => {
+                this.meshMap.set(frameNo, geometry)
+                resolve(true) // we only care about when this is resolved, to call callback()
+            })
+        })
+    }
+
+    decodeKTX2 = (textureURL: string, segmentNo: number) => {
+        return new Promise((resolve, reject) => {
+            this.ktx2Loader.load(textureURL, (texture: CompressedArrayTexture) => {
+                this.textureMap.set(segmentNo, texture)
+                resolve(true) // we only care about when this is resolved, to call callback()
+            })
+        })
+    }
+
+    processFrame = () => {
+        if (!this.fileHeader) {
+            return;
+        }
+
+        if (this.paused) {
+            /**
+             * Usually, this case arises when new track is set and fetchBuffers is still loading next frames.
+             * Until, startVideo is called, this.paused stays true.
+             */
+            this.onMeshBuffering(this.meshMap.size / this.geometryBufferSize)
+            return;
+        }
+
+
+        if (this.fileHeader.AudioURL) {
+            this.currentTime = this.audio.currentTime;
+        } else {
+            this.currentTime += this.clock.getDelta()
```

Comment:
We shouldn't use getDelta() here, we should just use Date.now() - playbackStartTime

Reply:
I was using [THREE.Clock](https://threejs.org/docs/#api/en/core/Clock) here, that uses perfomance.now internally when available. Shall I replace time handling with Date?